profiler/src/ProfilerEngine/Datadog.Profiler.Native.Windows/OsSpecificApi.cpp

```diff
@@ -239,4 +242,58 @@ int32_t GetProcessorCount()
     return nbProcs;
 }
 
+ScopedHandle GetThreadHandle(DWORD threadId)
+{
+    auto handle = ScopedHandle(::OpenThread(THREAD_QUERY_INFORMATION, FALSE, threadId));
+    if (handle == NULL)
+    {
+        LPVOID msgBuffer;
+        DWORD errorCode = GetLastError();
+
+        FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                      NULL, errorCode, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (LPTSTR)&msgBuffer, 0, NULL);
+
+        if (msgBuffer != NULL)
+        {
+            Log::Debug("GetThreadHandle: Error getting thread handle for thread id '", threadId, "': ", (LPTSTR)msgBuffer);
```

Comment:
FAR: handle the case if msgBuffer is NULL to still emit a warning with only the thread ID

Reply:
✔️